unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendsHUD/Scripts/FriendsHUDComponentView.cs

```diff
@@ -0,0 +1,230 @@
+    {
+        add => friendRequestsTab.OnRejectConfirmation += value;
+        remove => friendRequestsTab.OnRejectConfirmation -= value;
+    }
+    public event Action<string> OnFriendRequestSent
+    {
+        add => friendRequestsTab.OnFriendRequestSent += value;
+        remove => friendRequestsTab.OnFriendRequestSent -= value;
+    }
+    public event Action<FriendEntry> OnWhisper
+    {
+        add => friendsTab.OnWhisper += value;
+        remove => friendsTab.OnWhisper -= value;
+    }
+    public event Action<string> OnDeleteConfirmation
+    {
+        add => friendsTab.OnDeleteConfirmation += value;
+        remove => friendsTab.OnDeleteConfirmation -= value;
+    }
+    public event Action OnClose;
+
+    public RectTransform Transform => transform as RectTransform;
+    
+    public static FriendsHUDComponentView Create()
+    {
+        var view = Instantiate(Resources.Load<GameObject>("SocialBarV1/FriendsHUD")).GetComponent<FriendsHUDComponentView>();
+        return view;
+    }
+
+    public override void Awake()
+    {
+        base.Awake();
+        
+        friendsTabFocusButton.onClick.AddListener(() => FocusTab(0));
+        friendRequestsTabFocusButton.onClick.AddListener(() => FocusTab(1));
+        closeButton.onClick.AddListener(() =>
+        {
+            OnClose?.Invoke();
+            Hide();
+        });
+        
+        friendsTab.Expand();
+        friendRequestsTab.Expand();
+    }
+
+    public void HideSpinner()
+    {
+        loadingSpinner.SetActive(false);
+        model.isLoadingSpinnerActive = false;
+    }
+
+    public void ShowSpinner()
+    {
+        loadingSpinner.SetActive(true);
+        model.isLoadingSpinnerActive = true;
+    }
+
+    public List<FriendEntryBase> GetAllEntries()
+    {
+        var result = new List<FriendEntryBase>();
+        result.AddRange(friendsTab.Entries.Values);
+        result.AddRange(friendRequestsTab.Entries.Values);
+        return result;
+    }
+
+    public FriendEntryBase GetEntry(string userId)
+    {
+        return (FriendEntryBase) friendsTab.Get(userId) ?? friendRequestsTab.Get(userId);
+    }
+
+    public void UpdateEntry(string userId, FriendEntryBase.Model model)
+    {
+        friendsTab.Populate(userId, model);
+        friendRequestsTab.Populate(userId, model);
+    }
+
+    public void DisplayFriendUserNotFound() => friendRequestsTab.ShowUserNotFoundNotification();
+
+    public bool IsFriendListFocused() => friendsTab.gameObject.activeInHierarchy;
+
+    public bool IsFriendListCreationReady() => friendsTab.DidDeferredCreationCompleted;
+
+    public int GetReceivedFriendRequestCount() => friendRequestsTab.ReceivedRequestsList.Count();
+
+    public void Destroy() => Destroy(gameObject);
+
+    public void Show()
+    {
+        model.visible = true;
+        gameObject.SetActive(true);
+    }
+
+    public void Hide()
+    {
+        model.visible = false;
+        gameObject.SetActive(false);
+    }
```

Comment:
The `BaseComponentView` class already have a `Show` and `Hide` function implemented. You should override them if you want to introduce a new implementation.